Makefile

```
@@ -0,0 +1,40 @@
+
+MAKEFILE_ROOT=$(shell pwd)
+
+# install: Ensure that the Argo CD namespace exists, and that CRDs we are using are applied to the current cluster
+install:
+	kubectl create ns argocd || true
+	kubectl apply -f $(MAKEFILE_ROOT)/backend/config/crd/bases
+	kubectl apply -f $(MAKEFILE_ROOT)/backend-shared/config/crd/bases/managed-gitops.redhat.com_operations.yaml
+	kubectl apply -f https://raw.githubusercontent.com/argoproj/argo-cd/release-2.2/manifests/crds/application-crd.yaml
+
+# start: start all the components
+start:
+	~/go/bin/goreman start
+
+# build: build all the components (note: you do not need to do this before running start; 'start' will compile and run)
+build:
+	cd $(MAKEFILE_ROOT)/backend && make build
+	cd $(MAKEFILE_ROOT)/cluster-agent && make build
+
+# test: run tests for each component
+test:
+	cd $(MAKEFILE_ROOT)/backend && make test
+	cd $(MAKEFILE_ROOT)/backend-shared && make test	
+	cd $(MAKEFILE_ROOT)/cluster-agent && make test
```

```suggestion
test: test-backend test-backend-shared test-cluster-agent ## Run tests for all components

test-backend: ## Run tests for backend only
	cd $(MAKEFILE_ROOT)/backend && make test

test-backend-shared: ## Run test for backend-shared only
	cd $(MAKEFILE_ROOT)/backend-shared && make test

test-cluster-agent: ## Run test for cluster-agent only
	cd $(MAKEFILE_ROOT)/cluster-agent && make test
```

Same approach (reasoning) applies here as well.